File: TestInterop_paola/TestInterop_paola/Form1.vb
Imports System.Windows.Forms
Imports Word = Microsoft.Office.Interop.Word
Imports Excel = Microsoft.Office.Interop.Excel
Public Class Form1

    Private Sub btnGuardarWord_Click(sender As Object, e As EventArgs) Handles bttnGuardarWord.Click
        Dim dialogo As SaveFileDialog = New SaveFileDialog()

        If dialogo.ShowDialog() <> DialogResult.OK Then
            Return
        End If

        Dim ruta As String = dialogo.FileName
        Dim wordApp = New Word.Application()
        wordApp.Visible = True
        wordApp.Documents.Add()
        Dim dato As String = TextBox1.Text
        wordApp.Selection.TypeText(dato)
        wordApp.ActiveDocument.SaveAs2(ruta)
    End Sub

    Private Sub BtnGuardarExcel_Click(sender As Object, e As EventArgs) Handles bttnGuardarExcel.Click

        Dim dialogo As SaveFileDialog = New SaveFileDialog()
        If dialogo.ShowDialog() <> DialogResult.OK Then
            Return
        End If

        Dim dato As String = TextBox1.Text
        Dim ruta As String = dialogo.FileName
        Dim excelApp = New Excel.Application()
        excelApp.Visible = True
        excelApp.Workbooks.Add()
        Dim workSheet As Excel._Worksheet = CType(excelApp.ActiveSheet, Excel.Worksheet)
        workSheet.Cells(1, "A") = dato
        workSheet.Columns(1).AutoFit()
        workSheet.SaveAs(ruta)

    End Sub
End Class

File: TestInterop_paola/TestInterop_paolaCSharp/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Word = Microsoft.Office.Interop.Word;
using Excel = Microsoft.Office.Interop.Excel;

namespace TestInterop_paolaCSharp
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void TextBox1_TextChanged(object sender, EventArgs e)
        {

        }

        private void bttnGuardarWord_Click(object sender, EventArgs e)
        {
            SaveFileDialog dialogo = new SaveFileDialog();
            if (dialogo.ShowDialog() != DialogResult.OK)
            {
                return;
            }
            string ruta = dialogo.FileName;
            var wordApp = new Word.Application();
            wordApp.Visible = true;
            wordApp.Documents.Add();
            String dato = TextBox1.Text;

            wordApp.Selection.TypeText(dato);
            wordApp.ActiveDocument.SaveAs2(ruta);
        }

        private void label1_Click(object sender, EventArgs e)
        {

        }

        private void bttnGuardarExcel_Click(object sender, EventArgs e)
        {
            {
                SaveFileDialog dialogo = new SaveFileDialog();
                if (dialogo.ShowDialog() != DialogResult.OK)
                {
                    return;
                }
                string dato = TextBox1.Text;
                string ruta = dialogo.FileName;
                var excelApp = new Excel.Application();
                excelApp.Visible = true;
                excelApp.Workbooks.Add();
                Excel._Worksheet workSheet = (Excel.Worksheet)excelApp.ActiveSheet;
                workSheet.Cells[1, "A"] = dato;
                workSheet.Columns[1].AutoFit();

                workSheet.SaveAs(ruta);

            
             }
        }
    }
}




